### ISSUE TITLE
Review push notification doesn't open product review details when tapped

### ISSUE THREAD

**Issue by astralbodies** [woocommerce/woocommerce-ios#1810]
**Describe the bug**
When I received a push notification for a new product review, tapping it did not bring up the details of the review. I had product reviews up as the last screen I was using in the app before backgrounding it. The product review didn't appear in the list either until I pulled to refresh.

**To Reproduce**
1. Go to the Reviews tab.
2. Background the app.
3. Leave a product review to receive a push notification.
4. Tap the push notification to open the app.

**Screenshots**
Didn't take a screenshot as the text really covers it all.

**Expected behavior**
Tapping the push notification should force the review to be loaded and the details show up.

**Mobile Environment**
Please include:
- Device: iPhone 6s
- iOS version: 13.1.1
- WooCommerce iOS version: 3.5.0.0

**Comment by astralbodies** [comment]
Also the product name wasn't listed - only "Unknown".

**Comment by jaclync** [comment]
I noticed this issue in WCiOS 3.8.0.0 sometimes, this seemed to happen while the app has navigated to an Order or Review from a push notification before

**Comment by jaclync** [comment]
Re the comment above: I tried reproducing the issue by leaving the app in the background when it received the new order push notif, then receiving a new review push notif. Wasn't able to repro it this time though, it's probably caused by other reasons

**Comment by shiki** [comment]
I shared a similar scenario about order notifications in https://github.com/woocommerce/woocommerce-ios/issues/1867#issuecomment-639660324. I can reproduce the same for reviews. 

- It consistently opens the last screen only if the device was locked. 
- If the device was not locked, then the app opens the review details correctly. 

My device was also in low power mode.

**Comment by rachelmcr** [comment]
I saw this issue repeatedly while testing 5.4.0.0, exactly as described in the original issue (app was backgrounded but device wasn't locked, had to pull refresh the review list to see the new review, and product name wasn't listed).

I backgrounded the app once on an order detail screen and twice on the My store tab, and each time I got the same results.

**Comment by shiki** [comment]
We changed the way Product Review push notifications are handled in https://github.com/woocommerce/woocommerce-ios/pull/2669 so that the product will always be loaded. However, it looks like there was a regression in https://github.com/woocommerce/woocommerce-ios/pull/2879. If the product was not previously downloaded on the device, then the Review details page is never shown. 

In summary: There's a regression. 

---

New Order push notifications seem to be working though.

**Comment by shiki** [comment]
I also found another bug that happens when the app was previously terminated. The review details page will be shown but immediately removed from the view. 🙈 

1. Load all the products of the site.
1. Terminate the app.
2. Send a Product Review push notification. 
3. Tap on the push notification. 
4. The Review Detail page will be shown. But then it will be dismissed right away and the user will be navigated to the Reviews page.

**Comment by shiki** [comment]
☝️ The above was caused by https://github.com/woocommerce/woocommerce-ios/pull/2796 (5.1).

**Comment by designsimply** [comment]
I tested the steps provided in the original report and I was able to open review details as expected after tapping on a review push notification. 

Review Push Notification|Product Review detail opened after tapping push notification|New review not in list after pressing back arrow at top left|New review appeared after pull-to-refresh on the review list
---|---|---|---
![IMG_4780](https://user-images.githubusercontent.com/1119271/106823603-7d3bab00-663e-11eb-970f-b05a0c977e7e.PNG)|![IMG_4781](https://user-images.githubusercontent.com/1119271/106823606-7e6cd800-663e-11eb-857f-f8bb2ca51301.PNG)|![IMG_4782](https://user-images.githubusercontent.com/1119271/106823607-7e6cd800-663e-11eb-8415-17aab35c216f.PNG)|![IMG_4783](https://user-images.githubusercontent.com/1119271/106823608-7f056e80-663e-11eb-9671-2b3d67f7ef1e.PNG)

<sup>Tested with WCiOS 5.9.0.0 TestFlight beta, iPhone 11, iOS 14.4.</sup>

I was unable to see the new review in the reviews list until I did a pull-to-refresh but _only_ if the app was backgrounded. If the app was closed before receiving the push notification then the review list showed the review as expected. If the app was open then I received a snackbar notice and no push notification. I filed a separate issue for this at https://github.com/woocommerce/woocommerce-ios/issues/3584.

Recommend closing this issue if at least one other tester can no longer replicate the bug from the original report.

**Comment by hichamboushaba** [comment]
I confirm what @designsimply said, I can't reproduce the original issue, but can reproduce the new bug: the new review is missing from the review list if the app was backgrounded.

**Comment by joshheald** [comment]
This bug is still relevant: it may technically be new though.

Now that reviews have been moved to a tab on the Hub menu, I can't always get to the review details if I _wasn't_ already in the reviews section: it only takes me as far as the Hub menu.

Repro:

1. Open the app, navigate to any section outside of Reviews
2. On the web, write a review
3. Tap `View` when the in app notification is shown
4. Observe that you're taken to the review details 
5. Tap to another tab, and write another review on the web
6. Tap `View` when the notification is shown
7. Observe that you're only taken to the hub menu

The same happens when the app is backgrounded.

Once I got in to a situation where the `Reviews` option on the hub menu was non functional, and only force-closing the app would resolve that... but I couldn't get the app in to that state again.
